File: contests/abs/arc065_a.rs
use proconio::input;
use regex::Regex;

fn main() {
    input!(s: String);

    let cond = Regex::new(r"^(dream|dreamer|erase|eraser)+$").unwrap().is_match(&s);

    println!("{}", if cond { "YES" } else { "NO" });
}

File: contests/abs/arc065_a.fs
let bool f t cond = if cond then t else f

stdin.ReadLine()
|> fun s -> System.Text.RegularExpressions.Regex.IsMatch(s, "^(dream|dreamer|erase|eraser)+$")
|> bool "NO" "YES"
|> printfn "%s"
